environments/fetcher/fetcher.go

```diff
@@ -229,6 +240,76 @@ func (fetcher *Fetcher) Handler(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+func (fetcher *Fetcher) UploadHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, "", 405)
```

Comment:
please add a little error message there, like "only POST is supported on this endpoint".  Useful for debugging when things go wrong.